controller/metrics/metrics.go

```diff
@@ -136,6 +138,51 @@ func (m *MetricsServer) IncError(namespace, name string, kind string) {
 	}
 }
 
+// Remove removes the metrics server from the registry
+func (m *MetricsServer) Remove(namespace string, name string, kind string) {
+	go func(namespace string, name string, kind string) {
+		// wait for the metrics to be collected, prometheus scrape interval is 60 seconds by default
+		time.Sleep(defaults.GetMetricCleanupDelaySeconds())
+		switch kind {
+		case log.RolloutKey:
```

Comment:
don't we need to add the remaining keys for completeness https://github.com/argoproj/argo-rollouts/blob/f5e2d82cb3be3d0e7f51b35f81c8f34c0bbf929e/utils/log/log.go#L24-L28 ?

Reply:
We do have keep any metrics related to those so they would just be empty case statements.